editor/plugins/shader_editor_plugin.cpp

```diff
@@ -836,50 +837,224 @@ ShaderEditor::ShaderEditor() {
 	_editor_settings_changed();
 }
 
+void ShaderEditorPlugin::_update_shader_list() {
+	shader_list->clear();
+	for (uint32_t i = 0; i < edited_shaders.size(); i++) {
+		String text;
+		String path = edited_shaders[i].shader->get_path();
+		String _class = edited_shaders[i].shader->get_class();
+
+		if (path.is_resource_file()) {
+			text = path.get_file();
+		} else if (edited_shaders[i].shader->get_name() != "") {
```

Comment:
```suggestion
		} else if (!edited_shaders[i].shader->get_name().is_empty()) {
```
[AFAIK we settled on using this style.](https://github.com/godotengine/godot/pull/43181)

Reply:
I don't mind it, but I never agreed to it. I don't generally like to do this because it makes the code more difficult to read (! in one side and is_empy() on the other). Performance argument here makes little sense anyway.

Reply:
If you wanted to do this properly, I would probably add also an is_not_empty() function. We have this with `RID` and `Ref<>` (`is_null()` and `is_valid()`)

Reply:
I added a comment on #43181 in hopes to find a middle ground.